client/src/components/Forms/MarkdownInput.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { Input, Form, Button, Typography } from 'antd';
+import { useState } from 'react';
+import ReactMarkdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+
+export default function MarkdownInput(props: { [key: string]: any; notRequired?: boolean }) {
+  const { notRequired, ...otherProps } = props;
+  const [previewVisible, setPreviewVisible] = useState(false);
+  const [text, setText] = useState('');
+
+  const toogleView = () => {
+    setPreviewVisible(!previewVisible);
+  };
+
+  return (
+    <div style={{ marginBottom: '20px' }}>
+      <div style={{ display: previewVisible ? 'none' : 'block' }}>
+        <Form.Item
+          {...otherProps}
+          name="comment"
+          label="Comment"
+          rules={
+            notRequired
+              ? []
+              : [{ required: true, message: 'Please leave a detailed comment. Markdown syntax is allowed.', min: 30 }]
```

Comment:
I think not all 100% might know what is markdown. I suggest to add a link to markdown syntax guide (e.g. https://remarkjs.github.io/react-markdown/ or anything similar) above the input.

Reply:
Added link to: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

![Screenshot_15](https://user-images.githubusercontent.com/83158288/171999595-21a35d4d-b7cc-47c9-8a64-5debd3e77e41.png)

In addition to improved formatting, a big plus is the ability to insert images, which students lacked on a cross-check.

![Screenshot_14](https://user-images.githubusercontent.com/83158288/171999570-d8fddf51-3fbe-4eaf-85fe-97adca240b65.png)